src/stacks-hierarchy/Root_PurchaseOverviewScreen/components/TravellerSelection.tsx

```diff
@@ -27,65 +33,145 @@ export function TravellerSelection({
-  const selectableUserProfilesWithCount =
-    userCountState.userProfilesWithCount.filter((a) =>
-      selectableUserProfiles.some((b) => a.id === b.id),
-    );
-  useEffect(() => {
-    if (selectionMode == 'none' && selectableUserProfiles.length > 0) {
-      // If selectionMode is none, we should default to the first item.
-      // selectableUserProfiles should be only have one item, but this
-      // is specified by the fare product and reference data in firestore.
-      setTravellerSelection([
-        {
-          ...selectableUserProfiles[0],
-          count: 1,
-        },
-      ]);
-    } else {
-      const filteredSelection = userCountState.userProfilesWithCount.filter(
-        (u) => selectableUserProfiles.find((i) => i.id === u.id),
-      );
-      setTravellerSelection(filteredSelection);
-    }
-  }, [userCountState.userProfilesWithCount, fareProductType, selectionMode]);
+  const accessibility: AccessibilityProps = {
+    accessible: true,
+    accessibilityRole: 'button',
+    accessibilityLabel:
+      t(
+        selectionMode == 'multiple'
+          ? PurchaseOverviewTexts.travellerSelection.title_multiple
+          : PurchaseOverviewTexts.travellerSelection.title_single,
+      ) + screenReaderPause,
+    accessibilityHint: t(PurchaseOverviewTexts.travellerSelection.a11yHint),
+  };
+
+  const [selectableUserProfilesWithCount, setSelectableUserProfilesWithCount] =
+    useState<UserProfileWithCount[]>(selectableUserProfiles);
 
   useEffect(() => {
-    userCountState.updateSelectable(selectableUserProfiles);
-  }, [selectableUserProfiles]);
+    const filteredSelection = selectableUserProfilesWithCount.filter((u) =>
+      selectableUserProfiles.find((i) => i.id === u.id),
+    );
+    setTravellerSelection(filteredSelection);
+  }, [fareProductType, selectionMode, selectableUserProfilesWithCount]);
 
   if (selectionMode === 'none') {
     return <></>;
   }
 
+  let activePassengerCategoryCount = 0;
+  const totalPassengersCount = selectableUserProfilesWithCount.reduce(
+    (acc, sUPWC) => {
+      if (sUPWC.count > 0) {
+        activePassengerCategoryCount += 1;
+      }
+      return acc + sUPWC.count;
+    },
+    0,
+  );
+
+  const multiplePassengerCategoriesSelectedFrom =
+    activePassengerCategoryCount > 1;
+
+  const multiplePassengersDetailsText =
+    totalPassengersCount == 0
+      ? t(PurchaseOverviewTexts.travellerSelection.no_passengers_selected)
+      : selectableUserProfilesWithCount.reduce((acc, sUPWC) => {
+          if (sUPWC.count > 0) {
+            if (selectionMode == 'single') {
+              return getReferenceDataName(sUPWC, language);
+            }
+            if (acc !== '') {
+              acc += ', ';
+            }
+            return (
+              acc + sUPWC.count + ' ' + getReferenceDataName(sUPWC, language) // plural name values not ideal atm?
+            );
+          } else {
+            return acc;
+          }
+        }, '');
```

Comment:
I think this can be refactored to be more readable. Maybe first get an array of selected user profiles with a `filter` function. 

Then it might be something like (handwritten/pseudocode)
```ts
selectedUserProfiles
  .map(up => `${up.count} ${getReferenceDataName(up, language)}`)
  .join(', ')
```

This `selectedUserProfiles` may also be reused to get the `activePassengerCategoryCount`.

Reply:
Indeed 👍